vespajlib/src/main/java/com/yahoo/tensor/IndexedTensor.java

```diff
@@ -245,10 +253,16 @@ public static Builder of(TensorType type, DimensionSizes sizes) {
                                                        " but cannot be larger than " + size.get() + " in " + type);
             }
 
-            return new BoundBuilder(type, sizes);
+            if (type.valueType() == TensorType.Value.FLOAT)
+                return new IndexedFloatTensor.BoundFloatBuilder(type, sizes);
+            else if (type.valueType() == TensorType.Value.FLOAT)
```

Comment:
Same condition in if, should probably be ` == TensorType.Value.DOUBLE` here.

Reply:
Thanks! Functionally correct, but logically wrong ...